ReactNativeClient/lib/services/UndoRedoService.js

```diff
@@ -0,0 +1,34 @@
+function IObject() { this.i; return this; }
+const iObject = new IObject(); iObject.i = 0;
+const _iObject = new IObject(); _iObject.i = 0;
+let history = new Array();
+let text = '';
+
+class UndoRedoService {
+	static handleText(text) {
+		iObject.i++;
+		const y = iObject.i;
+		const x = text;
+		history[y] = x;
+	}
+
+	static getValue() { return text; }
+
+	static clearHistory() { history = []; }
+
+	static undo() {
+		if ((_iObject.i) < (iObject.i)) { _iObject.i++; }
+		const j = history.length - _iObject.i;
+
+		if (history[j]) { text = history[j]; } else { text = history[0]; }
+	}
+
+	static redo() {
+		if ((_iObject.i) > 1) { _iObject.i--; }
+		const j = history.length - _iObject.i;
+
+		if (history[j]) { text = history[j]; } else { text = history[0]; }
+	}
+}
+
+module.exports = UndoRedoService;
```

Comment:
I don't understand most of this code. It looks like code that's been minified, then un-minified, so please rewrite it using modern JavaScript standards (no `new Function`, no `new Array`, no variables named "x" or "i", Hungarian notation). Also, as per CONTRIBUTING.md, new classes should be written in TypeScript.

Reply:
Okay I will make the changes